src/apply_rules.rs

```diff
@@ -110,18 +209,137 @@ fn insert_code<T>(
         .insert((code.object_system, code.object_code));
 }
 
+fn update_prop<T: Clone + From<String> + Into<Option<String>>>(
+    p: &PropertyRule,
+    field: &mut T,
+    report: &mut Report,
+) {
+    let any_prop = Some("*".to_string());
+    if p.property_old_value == any_prop || p.property_old_value == field.clone().into() {
+        *field = T::from(p.property_value.clone());
+    } else {
+        report.add_warning(
+            format!(
+                "object_type={}, object_id={}, property_name={}: property_old_value does not match the value found in the data",
+                p.object_type.as_str(),
+                p.object_id,
+                p.property_name
+            ),
+            ReportType::OldPropertyValueDoesNotMatch,
+        );
+    }
+}
+
+fn wkt_to_geo(wkt: &str, report: &mut Report, p: &PropertyRule) -> Option<GeoGeometry<f64>> {
+    if let Ok(wkt) = wkt::Wkt::from_str(wkt) {
+        if let Ok(geo) = try_into_geometry(&wkt.items[0]) {
+            Some(geo)
+        } else {
+            warn!("impossible to convert empty point");
+            None
+        }
+    } else {
+        report.add_warning(
+            format!(
+                "object_type={}, object_id={}: invalid geometry",
+                p.object_type.as_str(),
+                p.object_id,
+            ),
+            ReportType::GeometryNotValid,
+        );
+        None
+    }
+}
+
+fn get_geometry_id(
+    wkt: &str,
+    collection: &mut CollectionWithId<Geometry>,
+    p: &PropertyRule,
+    report: &mut Report,
+) -> Option<String> {
+    if let Some(geo) = wkt_to_geo(wkt, report, p) {
+        let id = p.object_type.as_str().to_owned() + ":" + &p.object_id;
+        collection
+            .get_mut(&id)
+            .map(|mut g| {
+                g.geometry = geo.clone();
+            })
+            .unwrap_or_else(|| {
```

Comment:
wouldn't it be better to add something like https://doc.rust-lang.org/std/collections/hash_map/enum.Entry.html#method.or_insert_with for `CollectionWithId` ?
(but I don't know the best api to do this. @TeXitoi any idea?)
or maybe reuse https://github.com/CanalTP/navitia_model/blob/master/model-builder/src/builder.rs#L61 ?

Reply:
yes, get_or_create_with can be imported in CollectionWithId and use here and in the builder. An entry API is quite a lot of work.